packages/babel-plugin-transform-es2015-parameters/src/default.js

```diff
@@ -44,17 +50,33 @@ const iifeVisitor = {
 };
 
 export const visitor = {
-  Function(path) {
+  Function(path, state) {
     const { node, scope } = path;
     if (!hasDefaults(node)) return;
 
     // ensure it's a block, useful for arrow functions
     path.ensureBlock();
 
-    const state = {
-      iife: false,
-      scope: scope,
-    };
+    const params = path.get("params");
+
+    if (state.opts.loose) {
+      const body = [];
+      for (let i = 0; i < params.length; ++i) {
+        const param = params[i];
+        if (param.isAssignmentPattern()) {
+          body.push(buildLooseDefaultParam({
+            VARIABLE_NAME: param.get("left"),
```

Comment:
I copied over some more existing exec tests and realized the above is wrong because destructuring assignment is only executed when the default value is used, so I have to add an else clause. This led me to the following idea for the example above.

```javascript
function test(_temp) {
  {a: { b : c} = {}} = {} = _temp === undefined ? {} : _temp;
}
```

What do you think? I am testing this now.